external/storm-opentsdb/src/main/java/org/apache/stom/opentsdb/OpenTsdbMetricDatapoint.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.stom.opentsdb;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * This class represents a metric data point in OpenTSDB's format.
+ */
+public class OpenTsdbMetricDatapoint implements Serializable {
+
+    // metric name
+    private final String metric;
+
+    // map of tag value pairs
+    private final Map<String, String> tags;
+
+    // timestamp either in milliseconds or seconds at which this metric is occurred.
+    private final long timestamp;
+
+    // value of the metric
+    private final Object value;
+
+    private OpenTsdbMetricDatapoint() {
+        this(null, null, 0L, null);
```

Comment:
should null values be allowed?  This can result in NPE in equals/hashCode method